File: moxy-androidx-sample/src/main/kotlin/example/com/moxy_androidx_sample/BaseView.java
package example.com.moxy_androidx_sample;

import com.omegar.mvp.MvpView;
import com.omegar.mvp.viewstate.strategy.MoxyViewCommand;

import static com.omegar.mvp.viewstate.strategy.StrategyType.*;

public interface BaseView extends MvpView {

    @MoxyViewCommand(ADD_TO_END_SINGLE)
    void setTest(float test);

    float getTest();

    @MoxyViewCommand(ADD_TO_END)
    void testFunction();

}

File: app/src/main/java/com/omegar/mvp/base/BaseView.kt
package com.omegar.mvp.base

import com.omegar.mvp.Addon
import com.omegar.mvp.MvpView


/**
 * Created by Anton Knyazev on 09.06.2023.
 * Copyright (c) 2023 Omega https://omega-r.com
 */
interface BaseView<M>: MvpView {

    fun base(value: Addon<M>)

}